### PR TITLE
[MRG] Revert ptr_update_callback fixes segfaults

### LINKED ISSUE #408: extracellular calculations broken
It seems that #401 fixed the segmentation fault issue, but introduced a new one: **all potentials are zero**! My bad for not checking the example output...

sphx_glr_plot_record_extracellular_potentials_003-2

### PR DESCRIPTION
**Pull request by rythorpe**
Reverts #401 as suggested in #409.

Closes #408

### CODE REVIEW

File: hnn_core/extracellular.py [hunk 1]
```diff
@@ -504,16 +509,14 @@ def _build(self, cvode=None, include_celltypes='all'):
...
-                                                     1.))
+                                            h.Vector(n_total_segments, 1.))
 
         # record time for each array
         self._nrn_times = h.Vector().record(h._ref_t)
 
         # contributions of all segments on this rank to total calculated
         # potential at electrode (_PC.allreduce called in _simulate_dipole)
         self._nrn_voltages = h.Vector()
-        self._nrn_imem_vec = h.Vector(self.n_total_segments)
```
Comment (rythorpe) [inline hnn_core/extracellular.py:516]: Note that I took the liberty of deleting this line, as it is a duplicate of L482.

**Comment by jasmainak** [comment]
Thanks @rythorpe !

**Comment by jasmainak** [comment]
@cjayb you might want to take note of this. We still have the problem of segfault but that's an orthogonal issue.

**Comment by rythorpe** [comment]
As a matter of curiosity, I noticed that the [extracellular example](https://jonescompneurolab.github.io/hnn-core/dev/auto_examples/howto/plot_record_extracellular_potentials.html#sphx-glr-auto-examples-howto-plot-record-extracellular-potentials-py) on the master branch still doesn't show non-zero LFP recordings. Does it not update the documentation on master with that of the latest commit?

**Comment by jasmainak** [comment]
I can see the updated documentation. It probably took a while because the CI builds again on master branch after merge

**Comment by cjayb** [comment]
Thanks for taking care of this @rythorpe

**Pull request merged**